test/xpu_api/numerics/complex.number/complex.transcendentals/tanh.pass.cpp

```diff
@@ -61,19 +61,25 @@ void test_edges()
         }
         else if (std::isinf(testcases[i].real()) && std::isinf(testcases[i].imag()))
         {
+#if !_PSTL_TEST_COMPLEX_TANH_INF_INF_BROKEN
             assert(r.real() == (testcases[i].real() > 0 ? 1 : -1));
             assert(r.imag() == 0);
+#endif
         }
         else if (std::isinf(testcases[i].real()) && std::isnan(testcases[i].imag()))
         {
+#if !_PSTL_TEST_COMPLEX_TANH_MINUS_INF_NAN_BROKEN
             assert(r.real() == (testcases[i].real() > 0 ? 1 : -1));
             assert(r.imag() == 0);
+#endif
         }
         else if (std::isnan(testcases[i].real()) && testcases[i].imag() == 0)
         {
+#if !_PSTL_TEST_COMPLEX_TANH_NAN_MINUS_ZERO_BROKEN
             assert(std::isnan(r.real()));
             assert(r.imag() == 0);
             assert(std::signbit(r.imag()) == std::signbit(testcases[i].imag()));
+#endif
         }
```

Comment:
I'd skip the whole block (lines 62-83) instead; similarly in other tests.

Reply:
Disagree.
For each corner case in this test some separate check exist.
If we rewrite this part of code like
```C++
#if !_PSTL_TEST_COMPLEX_TANH_BROKEN
        else if (std::isinf(testcases[i].real()) && std::isinf(testcases[i].imag()))
        {
            assert(r.real() == (testcases[i].real() > 0 ? 1 : -1));
            assert(r.imag() == 0);
        }
        else if (std::isinf(testcases[i].real()) && std::isnan(testcases[i].imag()))
        {
            assert(r.real() == (testcases[i].real() > 0 ? 1 : -1));
            assert(r.imag() == 0);
        }
        else if (std::isnan(testcases[i].real()) && testcases[i].imag() == 0)
        {
            assert(std::isnan(r.real()));
            assert(r.imag() == 0);
            assert(std::signbit(r.imag()) == std::signbit(testcases[i].imag()));
        }
#endif
```
then we start other checks which has place in the code below.

Reply:
Fixed (where it's possible)